app/angular/src/client/preview/angular-beta/__testfixtures__/input.component.ts

```diff
@@ -0,0 +1,187 @@
+// @ts-nocheck
+/* eslint-disable no-console */
+/* eslint-disable no-underscore-dangle */
+import {
+  Component,
+  ElementRef,
+  EventEmitter,
+  HostBinding,
+  HostListener,
+  Input,
+  Output,
+  ViewChild,
+} from '@angular/core';
+
+export const exportedConstant = 'An exported constant';
+
+export type ButtonSize = 'small' | 'medium' | 'large' | 'xlarge';
+
+export enum ButtonAccent {
+  'Normal' = 'Normal',
+  'High' = 'High',
+}
+
+export interface ISomeInterface {
+  one: string;
+  two: boolean;
+  three: any[];
+}
+
+@Component({
+  selector: 'doc-button',
+  template: '<button>{{ label }}</button>',
+})
+export class InputComponent<T> {
```

Comment:
I think we're missing a test with 
```
  @Input('inputBindingPropertyName')
  public localPropertyName: string;
```
to complete the picture :) 
with your first version of code, it wouldn't have worked. now, I think it works without any problem